metaflow/plugins/argo/argo_workflows.py

```diff
@@ -839,6 +845,22 @@ def _container_templates(self):
             env_without_none_values = {k: v for k, v in env.items() if v is not None}
             del env
 
+            node_selector = []
+            if resources.get("node_selector"):
```

Comment:
Any reason for moving this code segment out from L:896?

Reply:
The problem is how Metaflow parses `node_selector`.

When set with the decorator (`@kubernetes(node_selector=["key=value", "foo=bar"])`), `resources.get("node_selector")` is an array of strings.
When set using the environment variable `METAFLOW_KUBERNETES_NODE_SELECTOR`, both `resources.get("node_selector")` and `KUBERNETES_NODE_SELECTOR` are strings like `key=value,foo=bar`
The previous code raises an exception:
```
IndexError: list index out of range
```

In addition, the code now raises an exception in case of malformed input. It took me a while to figure out what was the error

Reply:
@odracci we should ensure that the node_selector is properly formatted in `kubernetes_decorator.py` instead of this file. You would notice that this file doesn't do any validation of user-provided inputs for the decorators and assumes that that has already happened upstreams.

Reply:
@savingoyal it is not possible because `resources` is extracted on L:730 in this file. It requires a major refactor

Reply:
@odracci we should be able to set the `node_selector` attribute for the kubernetes decorator properly in L:97 of `kubernetes_decorator.py` and any necessary validation there or in `step_init`

Reply:
same issue described in this [comment](https://github.com/Netflix/metaflow/pull/1207/files#r1040209561)

Reply:
fixed